2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -103,10 +97,12 @@ function build_mongo_config() {
   member_id=1
   local container_addr
   container_addr="$(container_addr)"
+  local node
   for node in ${current_endpoints}; do
-    [ "$node" == container_addr ] && continue
-    members+="{ _id: ${member_id}, host: \"${node}:${CONTAINER_PORT}\"},"
-    let member_id++
+    if [ "$node" != "$container_addr" ]; then
```

Comment:
hook pods are not labeled to match the service, so they are not added as endpoints